src/nodejs/hooks/seeker_agent.go

```diff
@@ -0,0 +1,389 @@
+package hooks
+
+import (
+	"archive/zip"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/cloudfoundry/libbuildpack"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	EntryPointFile    = "SEEKER_APP_ENTRY_POINT"
+	agentDownloadPath = "/rest/api/latest/installers/agents/binaries/NODEJS"
+	SeekerRequire     = "require('./seeker/node_modules/@synopsys-sig/seeker');\n"
+)
+
+var pattern = regexp.MustCompile(`require\(['"].*@synopsys-sig/seeker['"]\)`)
+
+type SeekerCommand interface {
+	Execute(dir string, stdout io.Writer, stderr io.Writer, program string, args ...string) error
+}
+
+type Downloader interface {
+	DownloadFile(url, destFile string) error
+}
+type Unzipper interface {
+	Unzip(zipFile, absoluteFolderPath string) error
+}
+
+type SeekerAfterCompileHook struct {
+	libbuildpack.DefaultHook
+	Log                *libbuildpack.Logger
+	serviceCredentials *SeekerCredentials
+	Command            SeekerCommand
+	Downloader         Downloader
+	Unzzipper          Unzipper
+}
+
+type SeekerCredentials struct {
+	SeekerServerURL string
+}
+
+func init() {
+	logger := libbuildpack.NewLogger(os.Stdout)
+	command := &libbuildpack.Command{}
+	libbuildpack.AddHook(&SeekerAfterCompileHook{
+		Log:        logger,
+		Command:    command,
+		Downloader: SeekerDownloader{},
+		Unzzipper:  SeekerUnzipper{},
+	})
+}
+
+func (h SeekerAfterCompileHook) AfterCompile(compiler *libbuildpack.Stager) error {
+	h.Log.Debug("Seeker - AfterCompileHook Start")
+	c := SeekerCredentials{}
+	var err error
+	if c, err = extractServiceCredentialsUserProvidedService(h.Log); err != nil {
+		return err
+	}
+	if c == (SeekerCredentials{}) {
+		if c, err = extractServiceCredentials(h.Log); err != nil {
+			return err
+		}
+	}
+	if err = assertServiceCredentialsValid(c); err != nil {
+		return err
+	}
+	h.serviceCredentials = &c
+	credentialsJSON, _ := json.Marshal(h.serviceCredentials)
```

Comment:
Would like to see this error handled and not ignored. Please see: https://github.com/golang/go/wiki/CodeReviewComments#handle-errors

Reply:
fixed